ddtrace/propagation/http.py

```diff
@@ -885,12 +887,57 @@ def _inject(span_context, headers):
         return headers
 
 
+class _BaggageHeader:
+    @staticmethod
+    def _encode_key(key):
+        safe_characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'*+-.^_`|~"
+        encoded = urllib.parse.quote(key, safe=safe_characters)
+        return encoded
+
+    def _encode_value(value):
+        safe_characters = (
+            "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'()*+-./:<>?@[]^_`{|}~"
+        )
+        encoded = urllib.parse.quote(value, safe=safe_characters)
+        return encoded
+
+    @staticmethod
+    def _inject(span_context, headers):
+        baggage_items = span_context._baggage.items()
+        if not baggage_items:
+            return
+
+        header_value = ",".join(
+            f"{_BaggageHeader._encode_key(str(key).strip())}={_BaggageHeader._encode_value(str(value).strip())}"
```

Comment:
Should we raise an unhandled exception if value can not be converted to a string? How should we handle dicts,arrays,booleans,etc. The format needs to be consistent across languages (ex: encoding `True` vs `true`)

Reply:
When "extracting baggage from propagation headers, they may encounter malformed header contents." When this occurs, we "should ignore the entire header." So instead of an error, we could potentially just ignore it? As of now, I think it would just error.